implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java

```diff
@@ -210,15 +215,22 @@ private void initUnis(List<Uni<HealthCheckResponse>> list, Iterable<HealthCheck>
     }
 
     public void reportHealth(OutputStream out, SmallRyeHealth health) {
+        JsonObject payload = health.getPayload();
         if (health.isDown() && HealthLogging.logger.isInfoEnabled()) {
             // Log reason, as not reported by container orchestrators, yet container may get killed.
-            HealthLogging.logger.healthDownStatus(health.getPayload().toString());
+            HealthLogging.logger.healthDownStatus(payload.toString());
+        }
+
+        if (healthContentFilters != null) {
+            for (HealthContentFilter healthContentFilter : healthContentFilters) {
+                payload = healthContentFilter.filter(payload);
+            }
```

Comment:
```suggestion
        if (healthContentFilters != null) {
            for (Instance.Handle<HealthContentFilter> handle : healthContentFilters.handles()) {
                try {
                    payload = handle.get().filter(payload);
                } finally {
                    if (handle.getBean().getScope().equals(Dependent.class)) {
                        handle.destroy();
                    }
                }
            }
        }
```

This way you would get a leak if the content filter bean is declared as `@Dependent` - in which case you should explicitly destroy it after use

Reply:
You'll also need to add  `import jakarta.enterprise.context.Dependent` along with this

Reply:
@jmartisk thinking about this I don't think we should be destroying dependent filters. They are needed to filter every response for the duration of the SmallRyeHealthReporter which is application scoped. So I really don't think we should be destroying them here. And if we would destroy them when the AppScoped bean is destroyed that is when the application is closing, so I don't think it's needed. cc/ @mkouba

Reply:
@jmartisk is right. The problem is that every iteration over `Instance<HealthContentFilter>` would result in a new instance of a `@Dependent HealthContentFilter`. 

Now, if this filter defines a `@PreDestroy` callback then the injected `Instance<HealthContentFilter>` has to keep the reference until the `SmallRyeHealthReporter` is destroyed (at that time all the dependent filters would be destroyed as well).

Reply:
So, a little sorry goes to @jmartisk since I wasn't aware of this. Will update it everywhere. Thanks @mkouba.